spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/post/SendResponseFilter.java

```diff
@@ -210,10 +207,23 @@ private void addResponseHeaders() {
 		// Only inserts Content-Length if origin provides it and origin response is not
 		// gzipped
 		if (SET_CONTENT_LENGTH.get()) {
-			if (contentLength != null && !ctx.getResponseGZipped()) {
-				servletResponse.setContentLengthLong(contentLength);
+			if ( contentLength != null && !ctx.getResponseGZipped()) {
+				// To support Servlet API 3.0.1 we need to check if setcontentLengthLong exists
+				try {
+					HttpServletResponse.class.getMethod("setContentLengthLong");
```

Comment:
This will only occur if `zuul.set-content-length` is set to true, but I don't see another good way of doing this.